core_lib_generator/core_lib_generator_from_yaml.py

```diff
@@ -0,0 +1,111 @@
+import os
+import shutil
+
+import hydra
+
+from core_lib.helpers.string import camel_to_snake
+from core_lib_generator.generator_file_utils import add_imports_to_main_class
+from core_lib_generator.file_generators.data_access_generator import DataAccessGenerateTemplate, add_data_access_instances
+from core_lib_generator.file_generators.entity_generator import EntityGenerateTemplate
+from core_lib_generator.file_generators.jobs_generator import JobsGenerateTemplate
+
+hydra.core.global_hydra.GlobalHydra.instance().clear()
+hydra.initialize()
+
+
+class CoreLibGenerator:
+    def __init__(self, path_to_config: str):
+        config = hydra.compose(path_to_config)
+
+        self.core_lib_name = next(iter(config))
+        self.snake_core_lib_name = camel_to_snake(
+            self.core_lib_name
+        )  # make sure to use _to_safe_file_name() command line helpers
+        self.core_lib_config = config[self.core_lib_name].config
+
+        self.core_lib_entities = {}
+        self.core_lib_data_access = {}
+        self.core_lib_jobs = {}
+
+        if 'data' in config[self.core_lib_name].data_layers:
+            self.core_lib_entities = config[self.core_lib_name].data_layers.data
+            if 'data_access' in config[self.core_lib_name].data_layers:
+                self.core_lib_data_access = config[self.core_lib_name].data_layers.data_access
+                self.data_access_list = list(config[self.core_lib_name].data_layers.data_access.keys())
+        if 'jobs' in self.core_lib_config:
+            self.core_lib_jobs = self.core_lib_config.jobs
+
+    def _add_init_file(self):
+        folder_iter = os.walk(self.snake_core_lib_name)
+        for current_folder, _, _ in folder_iter:
+            open(f'{current_folder}/__init__.py', 'w').close()
+
+    def _generate_template(self, file_path: str, yaml_data: dict, template_generate):

```

Comment:
the parameter file_path is replaced by the content of `get_template_file `?

Reply:
`file_path` is the path of the new file in which we write the data returned from `generate()`
```
 with open(file_path, 'w') as file:
            file.write(new_file)
```